homeassistant/components/hegel/hegel_client.py

```diff
@@ -0,0 +1,284 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+from collections import deque
+from typing import Optional, Callable
+
+_LOGGER = logging.getLogger(__name__)
+
+CR = "\r"
+
+
+class HegelClient:
```

Comment:
I think this, along with a number of the consts should be moved to an external Python module.

Reply:
Yes, device specific code should be hosted in a python library